pysd/py_backend/vensim/vensim2py.py

```diff
@@ -313,6 +319,17 @@ def visit_imported_subscript(self, n, vc):
             args_str = vc[4]  # todo: make this less fragile?
             self.subscripts += get_external_data(f_str, args_str, root_path)
 
+        def visit_range(self, n, vc):
+            subs_start = vc[2].strip()
+            subs_end = vc[6].strip()
+            self.sequence, start, end = get_subscript_number_range(subs_start, subs_end)
```

Comment:
As the `get_subscript_number_range` function is only used once by `visit_range`, I would add the code here directly not making a separate function.

Reply:
I see your point but by leaving the function out of the visitor, I intend to simplify the complexity of the visitor code, following the principle: “When you feel the need to write a comment, first try to refactor the code so that any comment becomes superfluous."  ( Robert C. Martin's in "Clean code").

So, in my opinion, it is clearer and simpler to split it into a function with its documentation even though this function is only used once.

Reply:
I hear you, however, the structure of vensim2py is already complicated to follow. In other cases, unless functions from the builder are required, the necessary changes are done in place. I think this keeps the general structure of the vensim2py cleaner, nevertheless, the visitors are more loaded.

I leave it to @JamesPHoughton  to decide wether to use a separate function or not.

Reply:
This is a tricky one. The parser is a complicated piece of code, and I'm not sure of the best strategy for keeping it manageable. 

On the one hand, there are certainly going to be bits that are complex enough to be their own functions, even if they are only used once, and @marrobl's function documentation is super clean. On the other hand, the rest of the file localizes the code needed by the visitor in the visitor itself, and it's also going to be easier to understand code that picks a strategy and sticks to it. 

In the particular case of the parser, the style consistency argument is persuasive to me. I think putting the code all together in the visitor might be the best course. 

TBS, if we were going to rewrite the parser from scratch, or develop a brand new piece of code for PySD, I'd probably want to decompose things more and document functions better. This is the unfortunate legacy of the original developer not being a real software developer. ;) Sorry 'bout that...

Reply:
Okay, thank you both. I already put the code in the visitor